File: 349.Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        unordered_set<int> tmp(nums1.begin(), nums1.end());
        if (nums1.empty() || nums2.empty())
            return result;
        
        for (int i: nums2) {
            if (tmp.count(i)) {
                result.push_back(i);
                tmp.erase(i);
            }
        }
        
        return result;
    }
};

File: LeetCode/349.Intersection of Two Arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        return list(set(nums1) & (set(nums2)))

    def intersection2(self, nums1, nums2):
        record = set(nums1)
        result = set()
        for num in nums2:
            if num in record:
                result.add(num)
        return list(result)
